managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2172,6 +2176,32 @@ public long getCacheSize() {
         return entryCache.getSize();
     }
 
+    /**
+     * return BK error codes that are considered not likely to be recoverable
+     */
+    public static boolean isBkErrorNotRecoverable(int rc) {
+        switch (rc) {
+        case BKException.Code.NoSuchLedgerExistsException:
+        case BKException.Code.ReadException:
+        case BKException.Code.LedgerRecoveryException:
```

Comment:
This could be temporary, if multiple bookies are down and recovery is now failing, though it could succeed later.

Reply:
I think this may make sense to consider as `isBkErrorNotRecoverable` for ManagedCursor and not for ManagedLedger.? I think we can create two separate methods `isBkErrorNotRecoverable` for ManagedCursor and ManagedLedger as they may need different flexibility level?